prover/examples/zkmips.rs

```diff
@@ -268,14 +270,70 @@ fn prove_sha2_bench() {
     log::info!("private input value: {:X?}", private_input);
     state.add_input_stream(&private_input);
 
-    let (total_steps, mut state) = split_prog_into_segs(state, &seg_path, "", 0);
+    let (total_steps, mut state) = split_prog_into_segs(state, &seg_path, "", seg_size);
 
     let value = state.read_public_values::<[u8; 32]>();
     log::info!("public value: {:X?}", value);
     log::info!("public value: {} in hex", hex::encode(value));
 
-    let seg_file = format!("{seg_path}/{}", 0);
-    prove_single_seg_common(&seg_file, "", "", "", total_steps);
+    let mut seg_num = 1usize;
+    if seg_size != 0 {
+        seg_num = (total_steps + seg_size - 1) / seg_size;
+    }
+
+    if seg_num == 1 {
```

Comment:
Actually we intend to generate one segment and prove one.  But it's ok to generate multiple segments. if so, you dont have to check if seg_num is 1 or not, you can call  prove_multi_seg_common only.